File: 0383-ransom-note/0383-ransom-note.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        unordered_map<char, int>freq;

        for(char c : magazine) {
            freq[c]++;
        }

        for(char rn : ransomNote) {
            if(freq[rn] == 0) {
                return false;
            }else{
                freq[rn]--;
            }
        }
        return true;
    }
};

File: 0383-ransom-note/0383-ransom-note.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        Map<Character, Integer> map1 = new HashMap<>();

        for(char ch : magazine.toCharArray()) {
            map1.put(ch, map1.getOrDefault(ch, 0) + 1);
        }

        for(char ch : ransomNote.toCharArray()) {
            if(map1.containsKey(ch)) {
                map1.put(ch, map1.get(ch) - 1);

                if(map1.get(ch) == 0) {
                    map1.remove(ch);
                }
            }else {
                return false;
            }
        }
        return true;
    }
}